### DISCUSSION TITLE
Mesh

### DISCUSSION THREAD

**Discussion by rugric** [xcompact3d/Incompact3d#198] [Q&A] [answered]
Hello everyone. Is it possible to make the channel-flow restart with a different (denser) mesh than the one used in the first iterations?

**Comment by slaizet** [comment]
Hi,
Short answer: not automatically but you can do it manually by creating a small program to read the coarser restart file and writing a new restart file on the denser mesh.
Best,
Sylvain

**Reply by rugric** [reply]
If I modify the ux.bin, uy.bin, uz.bin files of the last save going to interpolate in the nodes of the new mesh could it be ok?

**Reply by pbartholomew08** [reply]
No, the restart data is read from the `checkpoint` file. You would need to adapt this reading to perform the interpolation.

**Reply by slaizet** [reply]
If you want to do a proper restart, you need to modify all the 3D arrays required for a proper restart.
Best,
Sylvain

**Reply by rugric** [reply]
Thank you very much for the tip. Once the 3D arrays in the checkpoint file are modified, the only thing to do is to set the right mesh parameters in the input file, right?

**Reply by slaizet** [reply, marked as answer]
yes, once all the arrays from the new checkpoint file have the correct size, you just need to change the mesh resolution on the input file.
Best,
Sylvain

**Comment by mathrack** [comment]
Hello,

It is probably easier to interpolate a snapshot and start the simulation with the interpolated velocity field.

Best,
Mathrack

Jun 26, 2023 12:28:17 rugric ***@***.***>:

**Reply by slaizet** [reply]
easier but not a proper restart :-)
